integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/TaskFhirResourceProviderIntegrationTest.java

```diff
@@ -0,0 +1,381 @@
+		assertThat(updatedTask.getAuthoredOn(), equalTo(authoredOn));
+		assertThat(task, validResource());
+		
+		// double-check the record returned via get
+		response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.JSON).go();
+		Task reReadTask = readResponse(response);
+		assertThat(reReadTask.getAuthoredOn(), equalTo(authoredOn));
+		
+	}
+	
+	@Test
+	public void shouldUpdateExistingTaskAsXml() throws Exception {
+		MockHttpServletResponse response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.JSON).go();
+		Task task = readResponse(response);
+		
+		// update the existing record
+		Date authoredOn = DateUtils.truncate(new Date(), Calendar.DATE);
+		task.setAuthoredOn(authoredOn);
+		
+		// send the update to the server
+		response = put("/Task/" + TASK_UUID).jsonContent(toJson(task)).xmlContext(toXML(task)).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		// read the updated record
+		Task updatedTask = readResponse(response);
+		
+		assertThat(updatedTask, notNullValue());
+		assertThat(updatedTask.getIdElement().getIdPart(), equalTo(TASK_UUID));
+		assertThat(updatedTask.getAuthoredOn(), equalTo(authoredOn));
+		assertThat(task, validResource());
+		
+		// double-check the record returned via get
+		response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.JSON).go();
+		Task reReadTask = readResponse(response);
+		
+		assertThat(reReadTask.getAuthoredOn(), equalTo(authoredOn));
+		assertThat(updatedTask.getStatus(), is(Task.TaskStatus.ACCEPTED));
+		
+	}
+	
+	@Test
+	public void shouldReturnBadRequestWhenDocumentIdDoesNotMatchTaskIdAsXML() throws Exception {
+		// get the existing record
+		MockHttpServletResponse response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.XML).go();
+		Task task = readResponse(response);
+		
+		// update the existing record
+		task.setId(WRONG_TASK_UUID);
+		
+		// send the update to the server
+		response = put("/Task/" + TASK_UUID).xmlContext(toXML(task)).go();
+		assertThat(response, isBadRequest());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+		
+	}
+	
+	@Test
+	public void shouldReturnNotFoundWhenUpdatingNonExistentTaskAsXML() throws Exception {
+		// get the existing record
+		// get the existing record
+		MockHttpServletResponse response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.XML).go();
+		Task task = readResponse(response);
+		
+		// update the existing record
+		task.setId(WRONG_TASK_UUID);
+		
+		// send the update to the server
+		response = put("/Task/" + TASK_UUID).xmlContext(toXML(task)).go();
```

Comment:
So, for the purposes of this test, we want to be "updating a non-existent task", but this is updating a Task that we *know* exists (since we just got it from the system). A proper implementation of this test would try to update a non-existent task UUID, e.g.,

```java
put("/Task/" + WRONG_TASK_UUID).xmlContext(toXML(task)).go();
```

Reply:
Resolved